src/runtime_src/core/common/drv/cu_plram.c

```diff
@@ -68,14 +75,15 @@ static void cu_plram_check(void *core, struct xcu_status *status)
 	 */
 	if (cu_plram->credits != cu_plram->max_credits)
 		done_reg = ioread32(cu_plram->vaddr + 0x1C);
-#endif
+out:
 	status->num_done = done_reg;
 	status->num_ready = done_reg;
 }
 
 static struct xcu_funcs xrt_cu_plram_funcs = {
 	.get_credit	= cu_plram_get_credit,
 	.put_credit	= cu_plram_put_credit,
+	.is_zero_credit = cu_plram_is_zero_credit,
```

Comment:
maybe you can name these API as: alloc_credit, free_credit, peek_credit (is_zero_credit seems to be too restricted).

Reply:
done